src/main/java/org/cloudfoundry/promregator/scanner/ReactiveAppInstanceScanner.java

```diff
@@ -50,103 +57,228 @@
 		private ResolvedTarget target;
-		
+
 		private String orgId;
 		private String spaceId;
 		private String applicationId;
-		
+		private String routeId;
+		private String domainId;
 		private String accessURL;
 		private int numberOfInstances;
+		private boolean internal;
+		private int internalPort;
+
 		/**
 		 * @return the target
 		 */
 		public ResolvedTarget getTarget() {
 			return target;
 		}
+
+		public String getDomainId() {
+			return domainId;
+		}
+
+		public void setDomainId(String domainId) {
+			this.domainId = domainId;
+		}
+
+		public String getRouteId() {
+			return routeId;
+		}
+
+		public void setRouteId(String routeId) {
+			this.routeId = routeId;
+		}
+
+		public int getInternalPort() {
+			return internalPort;
+		}
+
+		public void setInternalPort(int internalPort) {
+			this.internalPort = internalPort;
+		}
+
 		/**
 		 * @param target the target to set
 		 */
 		public void setTarget(ResolvedTarget target) {
 			this.target = target;
 		}
+
 		/**
 		 * @return the orgId
 		 */
 		public String getOrgId() {
 			return orgId;
 		}
+
 		/**
 		 * @param orgId the orgId to set
 		 */
 		public void setOrgId(String orgId) {
 			this.orgId = orgId;
 		}
+
 		/**
 		 * @return the spaceId
 		 */
 		public String getSpaceId() {
 			return spaceId;
 		}
+
 		/**
 		 * @param spaceId the spaceId to set
 		 */
 		public void setSpaceId(String spaceId) {
 			this.spaceId = spaceId;
 		}
+
 		/**
 		 * @return the applicationId
 		 */
 		public String getApplicationId() {
 			return applicationId;
 		}
+
 		/**
 		 * @param applicationId the applicationId to set
 		 */
 		public void setApplicationId(String applicationId) {
 			this.applicationId = applicationId;
 		}
+
 		/**
 		 * @return the accessURL
 		 */
 		public String getAccessURL() {
 			return accessURL;
 		}
+
 		/**
 		 * @param accessURL the accessURL to set
 		 */
 		public void setAccessURL(String accessURL) {
 			this.accessURL = accessURL;
 		}
+
+		/**
+		 * @return the isInternal
+		 */
+		public boolean isInternal() {
+			return internal;
+		}
+
+		/**
+		 * @param isInternal the isInternal to set
+		 */
+		public void setInternal(boolean isInternal) {
+			this.internal = isInternal;
+		}
+
 		/**
 		 * @return the numberOfInstances
 		 */
 		public int getNumberOfInstances() {
 			return numberOfInstances;
 		}
+
 		/**
 		 * @param numberOfInstances the numberOfInstances to set
 		 */
 		public void setNumberOfInstances(int numberOfInstances) {
 			this.numberOfInstances = numberOfInstances;
 		}
-		
+
 	}
-	
+
+	private Flux<AppRouteVector> determineRoutes(String orgId, String spaceId) {
+		Mono<ListRouteMappingsResponse> routeMappings = this.cfAccessor.retrieveRouteMappings();
+
+		Flux<RouteMappingResource> something = routeMappings
+				.flatMapMany(mapper -> Flux.fromIterable(mapper.getResources()));
+
+		Flux<AppRouteVector> routeMappingFlux = something.map(mapper -> {
+			AppRouteVector route = new AppRouteVector();
```

Comment:
Consider adjusting constructor here?

Reply:
Add the params to the constructor rather than setting them later?